crates/fuel-core/src/service/adapters.rs

```diff
@@ -83,19 +83,53 @@ impl P2PAdapter {
     }
 }
 
+#[cfg(feature = "p2p")]
+#[async_trait::async_trait]
+impl NetworkInfo for P2PAdapter {
+    async fn connected_reserved_peers(&self) -> usize {
+        if let Some(service) = self.service.as_ref() {
+            service.connected_reserved_peers_count().await.unwrap_or(0)
```

Comment:
Why would this error and if it does is zero appropriate?

Reply:
Only reason would be because of the `oneshot` channels failing - nothing else.
So while it shouldn't fail, if it does returning 0 should cause the check to be repeated again (after some delay)

Reply:
Well what is holding the other end of the oneshot because if that panics then the oneshot will continuously fail.

Reply:
I think Tommi's idea is that we want to know if it fails instead silently doing nothing. So maybe it is better to return `Result<usize, Err>` here and process the error on the `PoA` side by simple `?`. In this case the service will log this error, if something is wrong=)